keras_cv/models/object_detection/retina_net/retina_net.py

```diff
@@ -419,10 +442,16 @@ def _backward(self, y_true, y_pred):
 
         self.classification_loss_metric.update_state(classification_loss)
         self.box_loss_metric.update_state(box_loss)
-        self.regularization_loss_metric.update_state(regularization_loss)
+
+        if self._track_regularization:
+            self.regularization_loss_metric.update_state(regularization_loss)
         self.loss_metric.update_state(loss)
         return loss
 
+    @property
+    def _track_regularization(self):
```

Comment:
under what circumstances would `len(self.losses)` = 0?

Reply:
no regularization losses.  losses does not hold `compiled_loss`, only extra losses added via add_loss() & regularization.

Reply:
I see. makes sense to use as-is for now, but we should definitely fix it as Keras level.